sdk/lib/cxxrt/guard.cc

```diff
@@ -62,7 +66,7 @@ namespace
 			{
 				futex_wait(&high, LockBit);
 			}
-			assert(high == 0);
+			Debug::Assert(high == 0, "Corrupt guard word");
```

Comment:
It would be nice to log the address of the guard word here, so someone can look up that address. It will have a symbol name that matches the global being described.